src/Microsoft.Health.Fhir.Core/Features/Search/SearchOptionsFactory.cs

```diff
@@ -151,6 +161,29 @@ public SearchOptions Create(string resourceType, IReadOnlyList<Tuple<string, str
 
             options.UnsupportedSearchParams = unsupportedSearchParameters;
 
+            options.CompartmentType = null;
+
+            // Initialize enum to undefined value
+            var parsedCompartmentType = (CompartmentType)(-1);
+
+            if (!string.IsNullOrWhiteSpace(compartmentType) &&
+    !Enum.TryParse(compartmentType, out parsedCompartmentType))
+            {
+                throw new NotSupportedException(string.Format(Core.Resources.CompartmentTypeIsInvalid, compartmentType));
```

Comment:
You don't want to throw NotSupportedException since it's not derived from FhirException and which will result in 500. You probably want use InvalidSearchOperationException or a new exception type InvalidCompartmentSearchOperationException?